.github/workflows/checks.yaml

```diff
@@ -72,3 +72,24 @@ jobs:
             done
             exit 1
           fi
+
+  Check-Uncommitted-Make-All:
```

Comment:
I was imagining this would be a step inside the "Checks" job. Maybe after "Check go versions" and before "Test". Did you consider doing it that way? Are there advantages to having it in its own separate Job? (Not saying it must be like that, but I want to see what you think about it.)

Reply:
I thought having 2 scoped and self-explanatory tests (jobs) would be better than one long one. For visibility and UX. It allows you to see from the overview what failed without having to open the log. (I might as well remove the part that checks for uncommitted files from the first job).

Your suggestion has advantages as well, it's simpler and avoids the duplication of running `make all` separately.

I don't have a strong preference. In this case, I think the simpler option would be better.

Reply:
Oh, I've just noticed line 65-74 which looks like a very similar check. Is that one not working currently? Or does it catch some scenarios and miss others?

FWIW, since we have that already and it's implemented as a step in the existing job, I would suggest it's better to improve or fix that one, rather than create a new separate job with a similar step in it.

Reply:
+1

> Oh, I've just noticed line 65-74 which looks like a very similar check. Is that one not working currently? Or does it catch some scenarios and miss others?

This check works. The thing is that `make test` doesn't build everything. It misses "make docs".

Make targets:
```
test: manifests generate fmt vet
all: build manifests docs
build: generate fmt vet
```

Reply:
So we can keep the check, and just add `make all`

Reply:
Cool, I like it.

Reply:
Thanks @simonbaird for your feedback